sieve/example.py

```diff
@@ -1,13 +1,8 @@
 def sieve(limit):
-    primes = []
-
-    sv = [True] * (limit + 1)
-    sv[:2] = [False, False]     # 0 and 1 aren't prime
-
-    for n in range(limit + 1):
-        if sv[n]:  # n is prime
-            for i in range(n * 2, limit + 1, n):    # mark all multiples of n
-                sv[i] = False
-            primes.append(n)
-
+    primes = [2]
+    pRange = list(range(3, limit+1, 2))
+    while pRange:
+        p = pRange.pop(0)
+        pRange = list(filter(lambda x: x % p, pRange))
```

Comment:
It is returning a bool. Bool is a subclass of integers. I could make it clearer though.